src/Sentry/Protocol/Transaction.cs

```diff
@@ -0,0 +1,112 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.Json;
+using Sentry.Internal.Extensions;
+
+namespace Sentry.Protocol
+{
+    public class Transaction : ISpan, IJsonSerializable
```

Comment:
Transaction is actually a whole Event :(

That's how it is today in Sentry, basically an event with a `contexts.trace` field and spans

We can try to go with `ISpan` for now and see what it looks like.

Reply:
What is `contexts.trace` actually? I've seen it referenced before but still not sure.

Also, "transaction is actually a whole Event", does that affect how we should design the class? We probably can't make use of inheritance anyway due to manual serialization. I'm thinking of doing `public class Transactions : ISpan, IScope`, would  that be enough?

Reply:
we could inherit and call the base serializer to add some stuff and the Transaction one to add the rest.

In Java it was split in 2 parts, SentryEvent (current one) and Transaction, and both inherit from some BaseEvent.

Reply:
> we could inherit and call the base serializer to add some stuff and the Transaction one to add the rest.

This will be messy because it will have to add serialization logic before the `writer.WriteEndObject()`.

Reply:
we could call a `protected virtual AddMoreStuff(writer)` and implement this in the derived type